### PR TITLE
Staging

### PR DESCRIPTION
**Pull request by majestrate**

### CODE REVIEW

File: llarp/link/utp.cpp [hunk 1]
```diff
@@ -475,13 +489,13 @@ namespace llarp
...
       } while(true);
+#endif
     }
 #endif
 
     void
     LinkLayer::Pump()
     {
-      utp_issue_deferred_acks(_utp_ctx);
 #ifdef __linux__
```
Comment (neuroscr) [inline llarp/link/utp.cpp]: why only linux?

File: llarp/handlers/tun.cpp [hunk 2]
```diff
@@ -343,12 +343,6 @@ namespace llarp
         }
         else if(addr.FromString(qname, ".snode"))
         {
-          if(HasPathToSNode(addr.as_array()))
-          {
-            msg.AddINReply(ObtainIPForAddr(addr, true));
-            reply(msg);
-            return true;
-          }
```
Comment (neuroscr) [inline llarp/handlers/tun.cpp]: why did this get taken out?

File: llarp/ev/ev_kqueue.cpp [hunk 3]
```diff
@@ -374,21 +374,14 @@ llarp_kqueue_loop::tick(int ms)
       llarp::ev_io* ev = static_cast< llarp::ev_io* >(events[idx].udata);
       if(ev)
       {
-        if(events[idx].filter & EV_ERROR)
```
Comment (neuroscr) [inline llarp/ev/ev_kqueue.cpp]: why did we remove the EV_ERROR case?

File: llarp/link/utp.cpp (hunk 1 above)
Comment (majestrate) [reply llarp/link/utp.cpp]: icmp stuff is hard to do portably

File: llarp/handlers/tun.cpp (hunk 2 above)
Comment (majestrate) [reply llarp/handlers/tun.cpp]: the reply is deferred to when we have the ability to send to the service node now.

**Pull request merged**